bundles/Billing/Service/BillingService.php

```diff
@@ -245,6 +250,10 @@ public function putInvoice($invoiceData)
             throw new HttpException(404, 'Unable to update invoice with id '.$invoiceData['id'].' as it does not already exist');
         }
 
+        if ($invoice->getRecorded()) {
```

Comment:
Just out of curiosity: What does it mean that an invoice has been recorded?

Reply:
It means that the invoice has been finished and it can not be changed.